Held items

justinrreyes opened the pull request:
## Description

This PR modifies the InventoryManagement script slightly in order to handle displaying of selected items in front of the player camera. This is for enhanced clarity of what item the player is about to use/is carrying. 

## Type of Change

- [ ] Bug Fix
- [X] New Feature
- [X] Enhancement
- [X] Refactor
- [ ] Documentation
- [ ] Other (please specify):

## Related Issue

N/A

## Changes Made

- Modified InventoryManagement Script to allow displaying held items in front of player camera
- Modified Player Prefab with adding a transform for held items

## Checklist

- [X] Code follows the project's coding style.
- [X] If asset(s) added, they have been made into a prefab
- [X] Tests have been added/updated.
- [X] Documentation has been updated (if applicable).
- [X] No unused code or assets remain.
- [X] Unity project builds without errors.

## Screenshots (if applicable)

Holding Salt Item
![image](https://github.com/user-attachments/assets/46ebefa6-8f47-4590-aae4-6509992646ee)

Holding Healthpack
![image](https://github.com/user-attachments/assets/53a119f2-5b88-4f28-8a67-362f28a84022)

Holding Nothing
![image](https://github.com/user-attachments/assets/c2220f69-03dd-46df-9398-ef033e00b753)


## Additional Notes

Because of the small amount of items in the game I opted for a simpler approach to displaying items on screen. There are prefabs of each item attached to the player prefab and are disabled until the player selects an item in the hot bar, from there the correct item is then enabled to display it in front of the player camera. Once item is used or is no longer selected, item is disabled again.

This PR is based off of the Healthpack item PR and so the healthpack PR must be merged first

justinrreyes committed:
Implemented Displaying held Items

uid106 reviewed (approved):
## Pull Request Review
- **Functionality**:
  - Updated inventorymanagement to show items held in hand by the player
  - Modified Player Prefab with adding a transform for held items

- **Unused Code**: 
  - None

- **Code Style**: 
   - Properly following projects code style 

- **Script Structure**: 
   -  The structure of scripts is appropriate

- **Documentation**: 
   - Follows project documentation style

- **Build Integrity**: 
  - The project builds and runs as intended

---

### Summary
Double checked with all items and no items and it is functioning properly. The items show up clearly and equip correctly. One suggestion I have if it's not difficult is that the salt item looks a bit higher up in the hand in the vertical direction than the health pack. Your pictures look good in the PR, I think it looks slightly different than the pictures at 1920x1080. This is a minor thing I noticed, and not an issue to gameplay. This PR is ready to merge.

krain0 reviewed (approved):
## Pull Request Review
- **Functionality**:
  - **Held item previews:** These look good and work as intended. 
  - **Changes to inventory management:** This displays the appropriate item

- **Unused Code**:
  - None

- **Code Style**:
  - Fits with our conventions

- **Script Structure**:
  - Everything is well organized

- **Documentation**:
  - Provide clarification where needed

- **Build Integrity**:
  - The game built without issue on my machine.

---

### Summary
These item previews make it much easier to see which item is active. It works with all of the items we have added, and switching between them functions as intended. Looks ready to merge

uid106 committed:
Updated ghost mesh filter

Updated ghost mesh filter to fbx as Brandon couldn't see the .blend file on his computer

uid106 committed:
Merge branch 'main' into pr/50
